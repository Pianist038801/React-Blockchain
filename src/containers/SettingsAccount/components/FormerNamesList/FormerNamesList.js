import React, { Component } from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

class FormerNamesList extends Component {
  renderItem = (item, index) => {
    const { remove, pending } = this.props
    return (
      <tr key={index}>
        <td className="col-xs-12 col-md-2">{item.first_name}</td>
        <td className="col-xs-12 col-md-2">{item.last_name}</td>
        <td className="col-xs-12 col-md-2 text-center">
          {moment(item.start_date).format('DD/MM/YYYY')}
        </td>
        <td className="col-xs-12 col-md-2 text-center">
          {moment(item.end_date).format('DD/MM/YYYY')}
        </td>
        <td className="col-xs-12 col-md-3">
          <button
            className="btn red-btn action-btn"
            disabled={pending}
            onClick={() => {remove(item)}}
          >
            Remove
          </button>
        </td>
      </tr>
    )
  }

  render() {
    const { former_names } = this.props.user

    if (former_names.length === 0) {
      return null
    }

    return (
      <div className="form-group">
        <h2 className="form-group">Former names</h2>
        <table className="FormerNamesList VerificationList">
          <thead className="hidden-sm hidden-xs">
            <tr className="row">
              <th className="col-xs-12 col-md-2">First Name</th>
              <th className="col-xs-12 col-md-2">Last Name</th>
              <th className="col-xs-12 col-md-2 text-center">Start Date</th>
              <th className="col-xs-12 col-md-2 text-center">End Date</th>
              <th className="col-xs-12 col-md-3">Action</th>
            </tr>
          </thead>
          <tbody>
            {
              former_names.map(this.renderItem)
            }
          </tbody>
        </table>
      </div>
    )
  }
}

FormerNamesList.propTypes = {
  user: PropTypes.shape({
    former_names: PropTypes.array
  }),
  remove: PropTypes.func.isRequired,
  pending: PropTypes.bool.isRequired
}

export default FormerNamesList
