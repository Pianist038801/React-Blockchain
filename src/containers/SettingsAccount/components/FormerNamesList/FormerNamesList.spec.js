import React from 'react'
import renderer from 'react-test-renderer'
import FormerNamesList from './FormerNamesList'

describe('containers/SettingsAccount/components/FormerNamesList', () => {
  const props = {
    user: {
      former_names: [
        {first_name: 'jane', last_name: 'doe', start_date: '1990-01-01', end_date: '1997-01-01'},
        {first_name: 'john', last_name: 'doe', start_date: '1997-02-01', end_date: '2016-01-01'}
      ]
    },
    remove: jest.fn(),
    pending: false
  }

  describe('when former_names is not empty', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <FormerNamesList {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when former_names is empty', () => {
    it('renders correctly', () => {
      const props2 = {
        ...props,
        user: {
          former_names: []
        }
      }
      const tree = renderer.create(
        <FormerNamesList {...props2} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
