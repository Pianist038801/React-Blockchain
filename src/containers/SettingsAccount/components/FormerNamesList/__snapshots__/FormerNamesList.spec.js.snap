// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/SettingsAccount/components/FormerNamesList when former_names is empty renders correctly 1`] = `null`;

exports[`containers/SettingsAccount/components/FormerNamesList when former_names is not empty renders correctly 1`] = `
<div
  className="form-group"
>
  <h2
    className="form-group"
  >
    Former names
  </h2>
  <table
    className="FormerNamesList VerificationList"
  >
    <thead
      className="hidden-sm hidden-xs"
    >
      <tr
        className="row"
      >
        <th
          className="col-xs-12 col-md-2"
        >
          First Name
        </th>
        <th
          className="col-xs-12 col-md-2"
        >
          Last Name
        </th>
        <th
          className="col-xs-12 col-md-2 text-center"
        >
          Start Date
        </th>
        <th
          className="col-xs-12 col-md-2 text-center"
        >
          End Date
        </th>
        <th
          className="col-xs-12 col-md-3"
        >
          Action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="col-xs-12 col-md-2"
        >
          jane
        </td>
        <td
          className="col-xs-12 col-md-2"
        >
          doe
        </td>
        <td
          className="col-xs-12 col-md-2 text-center"
        >
          01/01/1990
        </td>
        <td
          className="col-xs-12 col-md-2 text-center"
        >
          01/01/1997
        </td>
        <td
          className="col-xs-12 col-md-3"
        >
          <button
            className="btn red-btn action-btn"
            disabled={false}
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
      <tr>
        <td
          className="col-xs-12 col-md-2"
        >
          john
        </td>
        <td
          className="col-xs-12 col-md-2"
        >
          doe
        </td>
        <td
          className="col-xs-12 col-md-2 text-center"
        >
          01/02/1997
        </td>
        <td
          className="col-xs-12 col-md-2 text-center"
        >
          01/01/2016
        </td>
        <td
          className="col-xs-12 col-md-3"
        >
          <button
            className="btn red-btn action-btn"
            disabled={false}
            onClick={[Function]}
          >
            Remove
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
